hs_collection_resource/models.py

```diff
@@ -34,6 +34,15 @@ def deleted_resources(self):
     def has_resources(self):
         return self.resources.count() > 0
 
+    @property
+    def are_all_contained_resources_published(self):
+        if not self.has_resources:
+            return False
+        for res in self.resources.all():
```

Comment:
Instead of the for loop, it may be better to write as follows:
    
    return self.resources.all().filter(published=True).exists()